renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -1198,7 +1243,7 @@ void QueryPrimitiveIdPerFragment(WrappedOpenGL *driver, GLReplay *replay,
 
     // we expect this value to be overwritten by the primitive id shader. if not
     // it will cause an assertion that all color values are the same to fail
-    driver->glClearColor(0.84f, 0.17f, 0.2f, 0.49f);
+    driver->glClearColor(0.0f, 0.17f, 0.2f, 0.49f);
```

Comment:
What's the reason for this change? As in the comment this is expected to be overwritten so is there a need to clear to 0?

Reply:
If that's the case then it's better to make sure the primitive ID gets set to `~0U` in error cases so that it's displayed in the UI as an unknown primitive. Returning 0 is actually worse than returning an obviously large and incorrect number, since 0 is a real primitive ID that will be present in every draw which could mean the user might well go down a wrong path based on bad information.